cpp/arcticdb/pipeline/write_frame.hpp

```diff
@@ -63,15 +62,19 @@ folly::Future<entity::AtomKey> append_frame(
         bool ignore_sort_order
 );
 
+enum class AffectedSegmentPart {
+    START,
+    END
+};
+
 std::optional<SliceAndKey> rewrite_partial_segment(
         const SliceAndKey& existing,
         IndexRange index_range,
         VersionId version_id,
-        bool before,
+        AffectedSegmentPart affected_part,
         const std::shared_ptr<Store>& store);
 
-std::vector<SliceAndKey> flatten_and_fix_rows(
-        const std::vector<std::vector<SliceAndKey>>& groups,
-        size_t& global_count);
+// TODO: Use std::span when C++20 is enabled
+std::vector<SliceAndKey> flatten_and_fix_rows(boost::span<const std::vector<SliceAndKey>> groups, size_t& global_count);
```

Comment:
Why `boost::span` and not just `array`?

Reply:
At the time of writing this I wasn't sure if we have variable length of elements. Looking at it now it might be the case that we always have 5 elements in it. Will update it.